execute/report/report.go

```diff
@@ -118,81 +121,144 @@ func buildSingleChainReport(
 		ProofFlagBits:       cciptypes.BigInt{Int: slicelib.BoolsToBitFlags(proof.SourceFlags)},
 	}
 
-	// Note: ExecutePluginReport is a strict array of data, so wrapping the final report
-	//       does not add any additional overhead to the size being computed here.
+	return finalReport, nil
+}
 
+type messageStatus string
+
+const (
+	ReadyToExecute  messageStatus = "ready_to_execute"
+	AlreadyExecuted messageStatus = "already_executed"
+	/*
+		SenderAlreadySkipped                 messageStatus = "sender_already_skipped"
+		MessageMaxGasCalcError               messageStatus = "message_max_gas_calc_error"
+		InsufficientRemainingBatchDataLength messageStatus = "insufficient_remaining_batch_data_length"
+		InsufficientRemainingBatchGas        messageStatus = "insufficient_remaining_batch_gas"
+		MissingNonce                         messageStatus = "missing_nonce"
+		InvalidNonce                         messageStatus = "invalid_nonce"
+		AggregateTokenValueComputeError      messageStatus = "aggregate_token_value_compute_error"
+		AggregateTokenLimitExceeded          messageStatus = "aggregate_token_limit_exceeded"
+		TokenDataNotReady                    messageStatus = "token_data_not_ready"
+		TokenDataFetchError                  messageStatus = "token_data_fetch_error"
+		TokenNotInDestTokenPrices            messageStatus = "token_not_in_dest_token_prices"
+		TokenNotInSrcTokenPrices             messageStatus = "token_not_in_src_token_prices"
+		InsufficientRemainingFee             messageStatus = "insufficient_remaining_fee"
+		AddedToBatch                         messageStatus = "added_to_batch"
+	*/
+)
+
+func (b *execReportBuilder) checkMessage(
+	_ context.Context, idx int, execReport plugintypes.ExecutePluginCommitDataWithMessages,
+	// TODO: get rid of the nolint when the error is used
+) (messageStatus, error) { // nolint this will use the error eventually
+	if slices.Contains(execReport.ExecutedMessages, execReport.Messages[idx].Header.SequenceNumber) {
+		return AlreadyExecuted, nil
+	}
+
+	return ReadyToExecute, nil
+}
+
+func (b *execReportBuilder) verifyReport(
```

Comment:
Notice `validationMetadata`, this type holds the encoded size and will also hold the gas limit and perhaps the fee boost ratio. It is computed per report, and accumulated as part of the builder.